bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java

```diff
@@ -57,58 +59,61 @@
  * @author Simon Merschjohann - Initial contribution
  */
 @Component(immediate = true)
+@NonNullByDefault
 public class RuleSupportScriptExtension implements ScriptExtensionProvider {
+
     private static final String RULE_SUPPORT = "RuleSupport";
     private static final String RULE_REGISTRY = "ruleRegistry";
     private static final String AUTOMATION_MANAGER = "automationManager";
 
-    private static Map<String, Collection<String>> presets = new HashMap<>();
-    private static Map<String, Object> staticTypes = new HashMap<>();
-    private static Set<String> types = new HashSet<>();
+    private static final Map<String, Collection<String>> PRESETS = new HashMap<>();
+    private static final Map<String, Object> STATIC_TYPES = new HashMap<>();
+    private static final Set<String> TYPES = new HashSet<>();
+
     private final Map<String, Map<String, Object>> objectCache = new ConcurrentHashMap<>();
 
-    private RuleRegistry ruleRegistry;
-    private ScriptedRuleProvider ruleProvider;
-    private ScriptedCustomModuleHandlerFactory scriptedCustomModuleHandlerFactory;
-    private ScriptedCustomModuleTypeProvider scriptedCustomModuleTypeProvider;
-    private ScriptedPrivateModuleHandlerFactory scriptedPrivateModuleHandlerFactory;
+    private @NonNullByDefault({}) RuleRegistry ruleRegistry;
+    private @NonNullByDefault({}) ScriptedRuleProvider ruleProvider;
+    private @NonNullByDefault({}) ScriptedCustomModuleHandlerFactory scriptedCustomModuleHandlerFactory;
+    private @NonNullByDefault({}) ScriptedCustomModuleTypeProvider scriptedCustomModuleTypeProvider;
+    private @NonNullByDefault({}) ScriptedPrivateModuleHandlerFactory scriptedPrivateModuleHandlerFactory;
```

Comment:
They seem to be mandatory so I think constructor injection would make sense.

Reply:
Done.